### DISCUSSION TITLE
NumberInput native event onChange

### DISCUSSION THREAD

**Discussion by joelroohzen** [mantinedev/mantine#3350] [Q&A]
### What package has an issue

@mantine/core

### Describe the bug

The NumberInput does not expose the native event to the passed down onChange handler - https://github.com/mantinedev/mantine/blob/efa63619c379ad709044b70ad40210030ec5bef0/src/mantine-core/src/NumberInput/NumberInput.tsx#L194

I can see in the comments in the top of this file this is done delibrately for the default controlled, but what is the rationale behind NOT passing the whole event up? 

I note that the TextInput does expose the event, so is there anyway to get it out of the NumberInput?

### What version of @mantine/hooks page do you have in package.json?

5.10.0

### Do you know how to fix the issue

None

### Are you willing to participate in fixing this issue and create a pull request with the fix

None

### Possible fix

if an onChange is passed, call it with the original event.

**Comment by rtivital** [comment]
NumberInput does not expose event because its value can also be changed by clicking up/down buttons which do not have such event.
